exist-core/src/main/java/org/exist/xquery/functions/map/MapType.java

```diff
@@ -94,6 +94,30 @@ public boolean contains(AtomicValue key) {
         return this.map.containsKey(key);
     }
 
+    @Override
+    public boolean containsReference(final Item item) {
+        for (final Iterator<Map.Entry<AtomicValue, Sequence>> it = map.iterator(); it.hasNext();) {
+            final Sequence value = it.next().getValue();
+            if (value == item) {
+                return true;
+            }
+            return value.containsReference(item);
```

Comment:
I guess this could have a problem, if there are more than one sequence entry.  Should may be written like so:
```
if (value == item || value.containsReference(item)) {
  return true;
}
```

Reply:
Okay fixed.